### PR TITLE
Userland: Add implementation of sponge

### PR DESCRIPTION
**Pull request by matthewgraham1**
Sponge is a command from the "moreutils" collection that soaks up all
standard input and outputs it to the desired file. This is a rewrite for
SerenityOS since it's such a simple and useful program. I named it 
"spng" because it is a lot more pleasant to type.

### CODE REVIEW

**Comment by matthewgraham1** [comment]
It looks like the auto builder doesn't like the license header I use. Is the 2-clause BSD license a requirement or is there some specific way that I should format the header?

**Comment by awesomekling** [comment]
It is absolutely a requirement yes :)

File: Userland/spng.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio wpath cpath rpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    auto in_file = Core::File::construct("/dev/stdin");
+    if (!in_file->open(Core::IODevice::ReadOnly)) {
```
Comment (bugaevc) [inline Userland/spng.cpp:41]: Please don't do it like that. We already have stdin opened (fd 0), just use that.

File: Userland/spng.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    auto output_buffer = in_file->read_all();
+
+    if (pledge("stdio wpath cpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    bool append = false;
+    const char* output_file_name = "/dev/stdout";
```
Comment (bugaevc) [inline Userland/spng.cpp:53]: Don't do that either :)

File: Userland/spng.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        perror("pledge");
+        return 1;
+    }
+
+    auto in_file = Core::File::construct("/dev/stdin");
+    if (!in_file->open(Core::IODevice::ReadOnly)) {
+        perror("open");
+        return 1;
+    }
+    auto output_buffer = in_file->read_all();
```
Comment (bugaevc) [inline Userland/spng.cpp:45]: It's not super awesome to read up all the data into RAM; the original sponge cat's the data to a temporary file, and later replaces the destination file with that temp file.

But I guess we can just go with reading everything for now

**Comment by matthewgraham1** [comment]
Sorry for the commit mess, I am new to git

**Comment by awesomekling** [comment]
This is not mergeable as-is, due to the commit mess you mention.

**Comment by awesomekling** [comment]
Closing this for now as there is no recent activity. If work on this resumes, feel free to reopen.

**Pull request closed without merging**